src/Notifier.js

```diff
@@ -0,0 +1,327 @@
+    const eventGood = !currentNotifieds.includes(id) && this.trackedEvents.includes(event);
+    const includesItems = items && items.length ? this.trackedRewards.some((r) => items.indexOf(r) >= 0) : true;
+    return eventGood && includesItems;
+  }
+
+  async notify (notifications) {
+    const usePush = true || !document.hasFocus();
+    notifications.forEach((notification) => {
+      if (usePush) {
+        this.notifier.show(notification.head, notification.body, {
+          onclick: (event) => {
+            if (notification.link) {
+              event.preventDefault();
+              window.open(notification.link, '_blank');
+            }
+          }
+        });
+
+        if (this.store.getters.sounds.includes(notification.type)) {
+          if (notification.sound === 'drum') {
+            const audio = new Audio(drum);
+            audio.volume = 0.2;
+            audio.play();
+          }
+          if (notification.sound === 'eidolon') {
+            const audio = new Audio(eidolon);
+            audio.volume = 0.2;
+            audio.play();
+          }
+        }
+
+      }
+    });
+  }
+
+  generateNotifications(ws) {
+    const toNotify = [];
+    // look for things to be notified
+    // run notifications for each
+    for (const alert of ws.alerts) {
+      if (this.isNotifiable(alert.id, 'alert', alert.rewardTypes)) {
+        toNotify.push(makeNotification('alert', alert));
+      }
+    }
+
+    for (const event of ws.events) {
+      if (this.isNotifiable(event.id, 'operation')) {
+        toNotify.push(makeNotification('operation', event));
+      }
+    }
+
+    if (ws.cetusCycle.shortString) {
+      if (ws.cetusCycle.isDay) {
+        if (this.isNotifiable(ws.cetusCycle.id, 'cetus.day')) {
+          toNotify.push(makeNotification('cetus.day', ws.cetusCycle));
+        }
+      } else {
+        if (this.isNotifiable(ws.cetusCycle.id, 'cetus.night')) {
+          toNotify.push(makeNotification('cetus.night', ws.cetusCycle));
+        }
+      }
+    }
+
+    const cetus = ws.syndicateMissions.filter((synd) => synd.syndicate === 'Ostrons')[0];
+    if (cetus && this.isNotifiable(cetus.id, 'syndicate.ostrons')) {
+      toNotify.push(makeNotification('syndicate.ostrons', cetus));
+    }
+
+    if (ws.voidTrader.active && this.isNotifiable(ws.voidTrader.id, 'baro')) {
+      toNotify.push(makeNotification('baro', ws.voidTrader));
+    }
+
+    for (const currentItem of ws.dailyDeals) {
+      if (this.isNotifiable(currentItem.id, 'darvo')) {
+        toNotify.push(makeNotification('darvo', currentItem));
+      }
+    }
+
+    for (const acolyte of ws.persistentEnemies) {
+      if (this.isNotifiable(acolyte.pid, 'enemies')) {
+        toNotify.push(makeNotification('enemies', acolyte));
+      }
+    }
+
+    if (ws.sortie && this.isNotifiable(ws.sortie.id)) {
+      toNotify.push(makeNotification('sortie', ws.sortie));
+    }
+
+    for (const fissure of ws.fissures) {
```

Comment:
In the future I would recommend finding a way to drastically reduce the cyclomatic complexity of this function in particular.

Reply:
I don't know of a good way. A similar but not same pattern has to be done for all the lists, and it would end up more complex by trying to generalize them, I think.